edgeless_api/proto/messages.proto

```diff
@@ -51,15 +57,15 @@ message SpawnFunctionRequest {
     StateSpecification state_specification = 4;
 }
 
-// Message to start a new resource instance.
-message StartResourceRequest {
-    // Resource type.
-    string class_type = 1;
-    // Configuration of the resource, which depends on the resource type.
-    map<string, string> configurations = 2;
+// Some APIs provide the same inteface for different id types (external vs internal)
+// This enables them to share a gRPC interface.
+message InstanceIdVariant {
+    oneof InstanceIdType {
```

Comment:
In general, I don't like particularly the use of oneof because of several reasons:

- it is poorly supported in C++ (I know this project is in Rust, but you never know where your gRPC interfaces will be used)
- it cannot be repeated
- unlike the other types, it does not have a well-defined default value. If you have an int, the default value is clear (0) and it will not be serialized on the wire. I like very much this feature of protobuf, which allows one to add many fields without worrying about overhead if not all are used. This is in stark contrast with most other protocol definition standards (e.g., ASN.1)

However, I understand that this specific case fits very well the intended usage and I will not object to keeping it.

If possible, I suggest limiting the future use of the `oneof` keyword.